paasta_tools/oom_logger.py

```diff
@@ -229,11 +262,18 @@ def main():
         container_id,
         process_name,
     ) in capture_oom_events_from_stdin():
-        try:
-            docker_inspect = client.inspect_container(resource_id=container_id)
-        except (APIError):
-            continue
-        env_vars = get_container_env_as_dict(docker_inspect)
+        if args.containerd:
+            # then we're using containerd to inspect containers
+            container_info = get_containerd_container(args.containerd, container_id)
+            container_spec_raw = container_info.spec.value.decode("utf-8")
+            container_inspect = json.loads(container_spec_raw)
+        else:
+            # we're using docker to inspect containers
+            try:
+                container_inspect = client.inspect_container(resource_id=container_id)
+            except (APIError):
+                continue
```

Comment:
We also probably want to mirror this try/except since containerd is likely to hit similar temporary issues (that said, it would probably be nice to log _something_ if we're dropping oom syslog messages and not processing them to paasta_oom_events here!)